schemachange/schemachange.c

```diff
@@ -631,6 +633,22 @@ int live_sc_post_add_int(struct ireq *iq, void *trans, unsigned long long genid,
         return 0;
     }
 
+    for (i = 0; i != maxblobs; ++i) {
+        bdb_get_compr_flags(iq->usedb->sc_from->handle, &oldodh, &reccompr,
+                            &oldcompr);
+        bdb_get_compr_flags(iq->usedb->sc_to->handle, &newodh, &reccompr,
+                            &newcompr);
+        (void)reccompr;
+
+        /* If we're removing the ODH, or changing the compression algorithm,
+           unpack the blobs. */
+        if ((oldodh && !newodh) || oldcompr != newcompr) {
+            rc = unodhfy_blob_buffer(iq->usedb, blobs + i, i);
+            if (rc != 0)
+                return rc;
+        }
+    }
+
```

Comment:
Consider unpacking blobs only after checking the stripe pointer so we dont do unnecessary work? Also consider making this logic into a function (so update can reuse)?

Reply:
Will do.